pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -164,19 +163,9 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		}
 
 		// Do not suspend a Pod whose owner is already managed by Kueue
-		if owner := metav1.GetControllerOf(pod.Object()); owner != nil {
-			if owner.Kind == "ReplicaSet" && owner.APIVersion == "apps/v1" {
-				// ReplicaSet is an implementation detail; skip over it to the user-facing framework
-				rs := &appsv1.ReplicaSet{}
-				err := w.client.Get(ctx, client.ObjectKey{Name: owner.Name, Namespace: pod.pod.GetNamespace()}, rs)
-				if err != nil {
-					return fmt.Errorf("failed to get replicaset: %w", err)
-				}
-				owner = metav1.GetControllerOf(rs)
-			}
-			if owner != nil && jobframework.IsOwnerIntegrationEnabled(owner) {
-				return nil
-			}
+		ancestorJob, err := jobframework.FindAncestorJobManagedByKueue(ctx, w.client, nil, pod.Object(), w.manageJobsWithoutQueueName)
```

Comment:
Actually, this change looks very suspicious. Because assume we have a pod group which is not managed by a parent inetgration. Shouldn't we still execute the LocalQueueDefaulting? Maybe this is not covered by e2e test, but we need to make sure this works ok with LQ defaulting.

Reply:
I think you're right - that's the current logic we have. 

https://github.com/kubernetes-sigs/kueue/blob/64ae56ba621d8f7acf122b77c77dfc017ffe530e/pkg/controller/jobs/pod/pod_webhook.go#L177-L179

I'm not sure we should fix it in this PR.

Reply:
I mean the case of plain pod - no owner - in that case we don't step into `if owner != nil && jobframework.IsOwnerIntegrationEnabled(owner) {` at all. Yet, in this case we would do the logic below for defaulting. Now, we skip the logic for defaulting as ancestorJob == nil. So, I think the PR in the current form regresses the case, unless I'm missing something.

Reply:
>  Now, we skip the logic for defaulting as ancestorJob == nil.

My mistake, sorry, if ancestorJob == nil the code will actually move forward to LQ defaulting as before.